src/dds/rs-dds-device-proxy.cpp

```diff
@@ -390,11 +390,19 @@ dds_device_proxy::dds_device_proxy( std::shared_ptr< const device_info > const &
     }
     set_matcher_type( matcher );
 
-    if (supports_info(RS2_CAMERA_INFO_PRODUCT_LINE) && 
-        !strcmp(get_info(RS2_CAMERA_INFO_PRODUCT_LINE).c_str(), "D500"))
-        set_auto_calibration_capability(std::make_shared<d500_auto_calibrated>(
-            std::make_shared<d500_debug_protocol_calibration_engine>(this)));
-            
+    if( supports_info( RS2_CAMERA_INFO_PRODUCT_LINE )
+        && ! strcmp( get_info( RS2_CAMERA_INFO_PRODUCT_LINE ).c_str(), "D500" ) )
+    {
+        // Find depth sensor to pass into d500_auto_calibrated object
+        sensor_base * depth_sensor = nullptr;
+        for( auto & sensor : sensor_name_to_info )
+            if( sensor.second.type == RS2_STREAM_DEPTH )
+                depth_sensor = sensor.second.proxy.get();
```

Comment:
What proxy means?
Is that for DDS only and not USB?

Reply:
What is the meaning to other D500 devices?

Reply:
Yes, proxy is for DDS devices.

DDS devices have the sensor on creation so we set it on the constructor. Other devices will set it only after creating the actual sensor.